src/category/service/hooks/useHandleAddCategory.tsx

```diff
@@ -0,0 +1,90 @@
+import useBottomSheet from '@/common-ui/BottomSheet/hooks/useBottomSheet';
+import { ChangeEventHandler, useCallback, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { v4 as uuid } from 'uuid';
+
+export interface CategoryItem {
+  id: string;
+  emoji: string;
+  categoryName: string;
+}
+
+const useHandleAddCategory = () => {
+  const router = useNavigate();
+  const onClickBack = () => router(-1);
+  // TODO : API 연동
+  const onClickSave = () => router(-1);
+
+  const [emoji, setEmoji] = useState('😎');
+  // 이모지 선택 모달에 가까움
+  // BS를 이용하여 이모지 라이브러리를 띄우면
+  // 프레임 드랍이 심하여 성능이 떨어짐
+  const {
+    close: setEmojiBSClose,
+    open,
+    isOpen: isEmojiBSOpen,
+  } = useBottomSheet();
+
+  const setEmojiBSOpen = useCallback(() => {
+    open();
+  }, []);
+
+  const onChangeEmoji = useCallback((emoji: string) => {
+    setEmoji(emoji);
+    setEmojiBSClose();
+  }, []);
+
+  const [categoryName, setCategoryName] = useState('');
+  const onChangeCategoryName = useCallback<
+    ChangeEventHandler<HTMLInputElement>
+  >((categoryName) => {
+    setCategoryName(categoryName.target.value);
+  }, []);
+
+  const isAllCategoryInfoFilled = (): boolean =>
+    Boolean(emoji.length && categoryName.length);
```

Comment:
```suggestion
  const isAllCategoryInfoFilled = !! (emoji.length && categoryName.length)
```

p4. 위와 같이 변수로 선언해줘도 괜찮지 않을까용??